bool grip
float64 effort
